docs/source/user-guide/arrow-introduction.md

```diff
@@ -0,0 +1,284 @@
+
+## Why Columnar? The Arrow Advantage
+
+Apache Arrow is an open **specification** that defines how analytical data should be organized in memory. Think of it as a blueprint that different systems agree to follow, not a database or programming language.
+
+### Row-oriented vs Columnar Layout
+
+Traditional databases often store data row-by-row:
+
+```
+Row 1: [id: 1, name: "Alice", age: 30]
+Row 2: [id: 2, name: "Bob",   age: 25]
+Row 3: [id: 3, name: "Carol", age: 35]
+```
+
+Arrow organizes the same data by column:
+
+```
+Column "id":   [1, 2, 3]
+Column "name": ["Alice", "Bob", "Carol"]
+Column "age":  [30, 25, 35]
+```
+
+Visual comparison:
+
+```
+Traditional Row Storage:          Arrow Columnar Storage:
+┌──────────────────┐              ┌─────────┬─────────┬──────────┐
+│ id │ name │ age  │              │   id    │  name   │   age    │
+├────┼──────┼──────┤              ├─────────┼─────────┼──────────┤
+│ 1  │  A   │  30  │              │ [1,2,3] │ [A,B,C] │[30,25,35]│
+│ 2  │  B   │  25  │              └─────────┴─────────┴──────────┘
+│ 3  │  C   │  35  │                   ↑          ↑         ↑
+└──────────────────┘              Int32Array StringArray Int32Array
+(read entire rows)                (process entire columns at once)
+```
+
+### Why This Matters
+
+- **Unified Type System**: All data sources (CSV, Parquet, JSON) convert to the same Arrow types, enabling seamless cross-format queries
+- **Vectorized Execution**: Process entire columns at once using SIMD instructions
+- **Cache Efficiency**: Scanning specific columns doesn't load unnecessary data into CPU cache
+- **Zero-Copy Data Sharing**: Systems can share Arrow data without conversion overhead
+
+Arrow has become the universal standard for in-memory analytics precisely because of its **columnar format**—systems that natively store or process data in Arrow (DataFusion, Polars, InfluxDB 3.0), and runtimes that convert to Arrow for interchange (DuckDB, Spark, pandas), all organize data by column rather than by row. This cross-language, cross-platform adoption of the columnar model enables seamless data flow between systems with minimal conversion overhead.
+
+Within this columnar design, Arrow's standard unit for packaging data is the **RecordBatch**—the key to making columnar format practical for real-world query engines.
+
+## What is a RecordBatch? (And Why Batch?)
+
+A **[`RecordBatch`]** cleverly combines the benefits of columnar storage with the practical need to process data in chunks. It represents a horizontal slice of a table, but critically, each column _within_ that slice remains a contiguous array.
+
+Think of it as having two perspectives:
+
+- **Columnar inside**: Each column (`id`, `name`, `age`) is a contiguous array optimized for vectorized operations
+- **Row-oriented outside**: The batch represents a chunk of rows (e.g., rows 1-1000), making it a manageable unit for streaming
```

Comment:
Maybe you can throw in here that the default batch size is 8192 rows and can be controlled by the `datafusion.execution.batch_size` config setting https://datafusion.apache.org/user-guide/configs.html